src/config/networks.js

```diff
@@ -27,145 +27,209 @@ const _DEFAULT_BACKEND_RULES = {
+  POOL_DEPOSIT: '500000000',
+  KEY_DEPOSIT: '2000000',
+}
+
+const JORMUNGANDR = {
+  PROVIDER_ID: YOROI_PROVIDER_IDS.JORMUNGANDR,
+  NETWORK_ID: NETWORK_REGISTRY.JORMUNGANDR,
+  MARKETING_NAME: 'Incentivized Testnet (ITN)',
+  ENABLED: false,
+  IS_MAINNET: false,
+  PROTOCOL_MAGIC: 764824073,
+  BACKEND: {
+    API_ROOT: 'https://shelley-itn-yoroi-backend.yoroiwallet.com/api',
+    ..._DEFAULT_BACKEND_RULES,
+  },
+  SEIZA_STAKING_SIMPLE: (ADA: string) =>
+    `https://testnet.seiza-website.emurgo.io/staking-simple/list?sortBy=RANDOM&searchText=&performance[]=0&performance[]=100&source=mobile&userAda=${ADA}`,
+  EXPLORER_URL_FOR_ADDRESS: (address: string) => `https://shelleyexplorer.cardano.org/address/?id=${address}`,
+  EXPLORER_URL_FOR_TX: (_tx: string) => {
+    throw new Error('non supported network')
+  },
+  LINEAR_FEE: {
+    CONSTANT: '200000',
+    COEFFICIENT: '100000',
+    CERTIFICATE: '400000',
+    PER_CERTIFICATE_FEES: {
+      CERTIFICATE_POOL_REGISTRATION: '500000000',
+      CERTIFICATE_STAKE_DELEGATION: '400000',
     },
   },
+  ADDRESS_DISCRIMINATION: {
+    PRODUCTION: '0',
+    TEST: '1',
+  },
+  GENESISHASH: '8e4d2a343f3dcf9330ad9035b3e8d168e6728904262f2c434a4f8f934ec7b676',
+  BLOCK0_DATE: 1576264417000,
+  SLOTS_PER_EPOCH: 43200,
+  SLOT_DURATION: 2,
+  PER_EPOCH_PERCENTAGE_REWARD: 19666,
+  BECH32_PREFIX: {
+    ADDRESS: 'addr',
+  },
+}
+
+// ALONZO
+const ALONZO_MAINNET = {
+  PROVIDER_ID: YOROI_PROVIDER_IDS.ALONZO_MAINNET,
+  ...HASKELL_SHELLEY,
+  MARKETING_NAME: 'Alonzo Main Net',
+}
+
+const ALONZO_TESTNET = {
+  PROVIDER_ID: YOROI_PROVIDER_IDS.ALONZO_TESTNET,
+  ...HASKELL_SHELLEY_TESTNET,
+  BACKEND: {
+    ...HASKELL_SHELLEY_TESTNET.BACKEND,
+    API_ROOT: 'https://alonzo-backend.yoroiwallet.com',
+  },
+  MARKETING_NAME: 'Alonzo Test Net',
+  EXPLORER_URL_FOR_ADDRESS: (address: string) =>
+    `https://explorer.alonzo-white.dev.cardano.org/address?address=${address}`,
+  EXPLORER_URL_FOR_TX: (tx: string) => `https://explorer.alonzo-white.dev.cardano.org/tx/${tx}`,
+}
+
+export const NETWORKS = {
+  // Deprecated
+  BYRON_MAINNET,
+  HASKELL_SHELLEY,
+  HASKELL_SHELLEY_TESTNET,
+  // Deprecated. Consider removing
+  JORMUNGANDR,
+}
+
+type NetworkConfig =
+  | typeof NETWORKS.BYRON_MAINNET
+  | typeof NETWORKS.HASKELL_SHELLEY
+  | typeof NETWORKS.HASKELL_SHELLEY_TESTNET
+  | typeof NETWORKS.JORMUNGANDR
+
+// PROVIDER
+// NOTE: For now we are using the same obj, later we will decouple network/protocol/connection
+export const PROVIDERS = {
+  BYRON_MAINNET,
+  HASKELL_SHELLEY,
+  HASKELL_SHELLEY_TESTNET,
+  ALONZO_MAINNET,
+  ALONZO_TESTNET,
+  JORMUNGANDR,
+}
+
+export type YoroiProviderConfig =
+  | typeof BYRON_MAINNET
+  | typeof HASKELL_SHELLEY
+  | typeof HASKELL_SHELLEY_TESTNET
+  | typeof ALONZO_MAINNET
+  | typeof ALONZO_TESTNET
+  | typeof JORMUNGANDR
+
+export const getYoroiProvider = (networkConfig: NetworkConfig, provider: ?YoroiProvider): YoroiProviderConfig => {
```

Comment:
why is this called YoroiProvider and not just Provider? or maybe CardanoProvider?

Reply:
In a long term, we will be building yoroi for many targets i'd like to start identifying the yoroi interfaces by prefixing them, therefore when we have many implementations for ex. to Address we know it will have to implement YoroiAddress.